class-zoninator-rest-api-controller.php

```diff
@@ -0,0 +1,575 @@
+                'post_type' => $post_types,
+                'ignore_sticky_posts' => true,
+                'post_status' => array( 'publish', 'future' ),
+                'post__not_in' => $zone_post_ids,
+            ) );
+
+            if ( $this->_validate_category_filter( $cat ) ) {
+                $args['cat'] = $cat;
+            }
+
+            if ( $this->_validate_date_filter( $date ) ) {
+                $filter_date_parts = explode( '-', $date );
+                $args['year'] = $filter_date_parts[0];
+                $args['monthnum'] = $filter_date_parts[1];
+                $args['day'] = $filter_date_parts[2];
+            }
+
+            $content = '';
+            $recent_posts = get_posts( $args );
+            foreach ( $recent_posts as $post ) {
+                $content .= sprintf('<option value="%d">%s</option>', $post->ID, get_the_title($post->ID) . ' (' . $post->post_status . ')');
+            }
+
+            wp_reset_postdata();
+            $status = 1;
+        }
+
+        if ( ! $content ) {
+            $empty_label = __( 'No results found', self::ZONINATOR_NAMESPACE );
+        } elseif ( $cat ) {
+            $empty_label = sprintf(__('Choose post from %s', 'zoninator'), get_the_category_by_ID($cat));
+        } else {
+            $empty_label = __( 'Choose a post', self::ZONINATOR_NAMESPACE );
+        }
+
+        $content = '<option value="">' . esc_html($empty_label) . '</option>' . $content;
+
+        $response = new WP_REST_Response(array(
+            'zone_id' => $zone_id,
+            'content' => $content,
+            'status' => $status));
+        $response->set_status($http_status);
+        return $response;
+    }
+
+    /**
+     * Check if a given request has access to get items.
+     *
+     * @param WP_REST_Request $request Full data about the request.
+     * @return WP_Error|bool
+     */
+    public function get_items_permissions_check( $request ) {
+        return true;
+    }
+
+    /**
+     * Check if a given request has access to delete item.
+     *
+     * @param WP_REST_Request $request Full data about the request.
+     * @return WP_Error|bool
+     */
+    public function delete_item_permissions_check( $request ) {
+        return $this->verify_access( $request );
+    }
+
+    /**
+     * Check if a given request has access to create item.
+     *
+     * @param WP_REST_Request $request Full data about the request.
+     * @return WP_Error|bool
+     */
+    public function create_item_permissions_check( $request ) {
+        return $this->verify_access( $request );
+    }
+
+    /**
+     * Check if a given request has access to update item.
+     *
+     * @param WP_REST_Request $request Full data about the request.
+     * @return WP_Error|bool
+     */
+    public function update_item_permissions_check( $request ) {
+        return $this->verify_access( $request );
+    }
+
+    public function verify_access(WP_REST_Request $request) {
```

Comment:
Do we need this intermediary method? We could just have the `*_permissions_check` methods call out to `_permissions->check` directly and skip the added complexity from the `switch`.